### ISSUE TITLE
 Z-Offset not being calibrated in the middle of the build platform

### ISSUE THREAD

**Issue by steveruss** [Robo3D/Customer_Feedback#9]
I have 2 R2 printers running R2 firmware 1.8.1-r2.1

When I run the Utilities/Wizards/Z-Offest it computes the Z-Offset at the extreme front left of the print bed:

![z-offset](https://user-images.githubusercontent.com/6875909/33906667-d347c09e-df3f-11e7-8a47-03557d325719.JPG)

...and not at the center of the print bed where the prints will be made.

This makes absolutely NO sense at all. The Z is being set OUTSIDE of the bed leveling adjustment screws so any difference between the bed leveling screws is being projected to create an even greater error.

When the Z is calculated in the center of the bed...any difference between these adjustment settings would be averaged instead.

Here are the instructions from your docs website showing the Z-Offset wizard using the center of the print bed.

https://help.robo3d.com/hc/en-us/articles/115000712832-Calibration-and-setting-Z-Offset

Why is this different on both of my R2s?

Is this a bug in the firmware?

How does someone get this wizard to work like is shown in the docs?

**Comment by OutsourcedGuru** [comment]
Just noting that the sensor is at the back of the carriage assembly rather than the front or middle of it.

**Comment by peries** [comment]
@steveruss thanks for this report. we changed the wizard to calibrate the z-offset to the front corner as this is the 0,0 origin and the main point of reference used to create the mesh of the bed slope. through testing we found this to give a more accurate z-offset value.

our online documentation has been updated to reflect this change. 
[C2](https://help.robo3d.com/hc/en-us/articles/115000715211-Calibration-and-setting-Z-Offset)
[R2](https://help.robo3d.com/hc/en-us/articles/115000712832-Calibration-and-setting-Z-Offset)

thanks!
